tsconfig.json

```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "noEmit": true,
+    "module": "commonjs",
+    "target": "ES2017",
+    "allowJs": true,
+    "checkJs": true,
+    "strict": true,
```

Comment:
// "listFiles": true,

add that so its easier to verify what was checked by TSC. 

for the curious, here's what's being checked now: 
![image](https://user-images.githubusercontent.com/39191/33050637-bffc0cc4-ce1a-11e7-9538-1275be8882e0.png)

we can't flip it on for all of `lib/*` but ` "lighthouse-core/gather/*.js"` is currently fine.